### PR TITLE
Add hot-reloading

### PR DESCRIPTION
**Pull request by chrisza4**
Add hot reload. Web server will reflect any code changes automatically and tighten feedback loop.

Also:
- Improve documentation
- Add basic route

### CODE REVIEW

File: web/rest-crud-demo/README.md [hunk 1]
```diff
@@ -4,13 +4,16 @@ FIXME: description
...
+- etc (TBD)
 
 ## Usage
 
-FIXME: explanation
+Running in local
 
-    $ java -jar rest-crud-demo-0.1.0-standalone.jar [args]
+```
+lein start
```
Comment (jpaulorio) [inline web/rest-crud-demo/README.md:45]: lein start didn't work for me. lein run works. am I missing something?

Comment (jpaulorio) [reply web/rest-crud-demo/README.md:45]: also, I'm able to run locally using lein ring server-headless. any thoughts on what's the best approach?

Comment (chrisza-tw) [reply web/rest-crud-demo/README.md:45]: TBH, I am not sure which is better.

Comment (zigolab) [reply web/rest-crud-demo/README.md:45]: Is there any chance to not use lein? In the cli subproject we tried to stick to deps.edn, and I'm curious to see if and how we can implement hot-reloading with it...

**Pull request merged**